.github/workflows/check-pr-branch.yml

```diff
@@ -0,0 +1,18 @@
+name: Disallow Pull Requests to Master Branch
```

Comment:
What do you think about making the name something that suggests the appropriate action? Like "Pull Requests should target Staging branch", so that hopefully people can self-diagnose without having to click through to the logs